## Problem Set 1

In [18]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [19]:
#1a
df = pd.read_csv("/Users/zhengyaojin/Desktop/charitable_giving.csv")
df.describe()

donation_amount  donation_dummy       control     treatment  \
count     50083.000000    50083.000000  50083.000000  50083.000000   
mean          0.915694        0.020646      0.333187      0.666813   
std           8.709199        0.142197      0.471357      0.471357   
min           0.000000        0.000000      0.000000      0.000000   
25%           0.000000        0.000000      0.000000      0.000000   
50%           0.000000        0.000000      0.000000      1.000000   
75%           0.000000        0.000000      1.000000      1.000000   
max         400.000000        1.000000      1.000000      1.000000   

        match_ratio        ratio1        ratio2        ratio3  \
count  50083.000000  50130.000000  50083.000000  50083.000000   
mean       1.333546      0.222083      0.222311      0.222211   
std        1.154606      0.415651      0.415803      0.415736   
min        0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000   
50%        1.000000      0.000000      0.000000      0.000000   
75%        2.000000      0.000000      0.000000      0.000000   
max        3.000000      1.000000      1.000000      1.000000   

       red_state_dummy  months_since_last_donation  highest_previous_donation  \
count     50048.000000                50082.000000               50083.000000   
mean          0.404452                   13.007268                  59.384977   
std           0.490791                   12.081403                  71.177310   
min           0.000000                    0.000000                   0.000000   
25%           0.000000                    4.000000                  30.000000   
50%           0.000000                    8.000000                  45.000000   
75%           1.000000                   19.000000                  60.000000   
max           1.000000                  168.000000                1000.000000   

       prior_donations_num  
count         50083.000000  
mean              8.039355  
std              11.394454  
min               0.000000  
25%               2.000000  
50%               4.000000  
75%              10.000000  
max             218.000000

In [42]:
df.dropna(subset=['months_since_last_donation'], inplace=True)
X = df['treatment']
X = sm.add_constant(X)
y = df['months_since_last_donation']
model = sm.OLS(y, X)
result = model.fit()
print(result.summary())

                                OLS Regression Results                                
Dep. Variable:     months_since_last_donation   R-squared:                       0.000
Model:                                    OLS   Adj. R-squared:                 -0.000
Method:                         Least Squares   F-statistic:                   0.01428
Date:                        Sat, 27 Jan 2024   Prob (F-statistic):              0.905
Time:                                16:27:53   Log-Likelihood:            -1.9585e+05
No. Observations:                       50082   AIC:                         3.917e+05
Df Residuals:                           50080   BIC:                         3.917e+05
Df Model:                                   1                                         
Covariance Type:                    nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

Intrepretation: For individuals who are in the treatment group (treatment = 1), there is an average of 12.998 + 0.014 = 13.012 months since the last donation. For individuals who are in the control group (control = 1), there is an estimated average of 12.998 months since the last donation.


The coefficients in the summary output above align with the group-specific means reported in Table 1 of the paper. 

1(b): Since the p-value for the treatment coefficient is 0.905 and we are testing the significance at the 95% confidence level, we fail to reject the null hypothesis that there's no difference in number of month since last donation between treatment and control as 0.905 is clearly bigger than 0.05. Hence, it is not statisically significant. The result is expected because it indicates that the pattern of contribution between two groups are roughly the same, showing that the randomization procedure is done correctly. This would add credibility to the experimental results since there is no such bias like historical contribution pattern existed. 

1(c). Table 1 shows that the values between control and treatment regarding some metrics like member activity, census demographic, and state and country organizational activity are all very similar to each other. This may well indicate that the experiment took into account the difference between different people demogrpahics and nation wide demogrpahics and use randomization to establish comparable and less-biased treatment and control group. 

In [34]:
#2a
X_1 = df['treatment']
X_1 = sm.add_constant(X_1)
y_1 = df['donation_dummy']
model_1 = sm.OLS(y_1, X_1).fit()
print(model_1.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     9.621
Date:                Sat, 27 Jan 2024   Prob (F-statistic):            0.00192
Time:                        15:05:56   Log-Likelihood:                 26629.
No. Observations:               50082   AIC:                        -5.325e+04
Df Residuals:                   50080   BIC:                        -5.324e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0179      0.001     16.224      0.0

Individuals who are in the treatment group are estimated to have 0.0042 higher likelihood of making a donation than individuals who are in the control group, holding all other variables constant. Individuals who are in the control group are estimated to have 0.018 likelihood of making a donation. These align with the results in the paper (control 0.018, treatment 0.022). 

In [35]:
#2b
X_2 = df[['ratio1', 'ratio2','ratio3']]
X_2 = sm.add_constant(X_2)
y_2 = df['donation_dummy']
model_2 = sm.OLS(y_2, X_2).fit()
print(model_2.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.666
Date:                Sat, 27 Jan 2024   Prob (F-statistic):             0.0117
Time:                        15:08:47   Log-Likelihood:                 26629.
No. Observations:               50082   AIC:                        -5.325e+04
Df Residuals:                   50078   BIC:                        -5.322e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0179      0.001     16.224      0.0

With 0 matching, which is the intercept and control group in this case, people have 1.79 percent likelihood of responding to the donation. 
With 1:1 matching, people have 0.29 percent more likelihood of responding to the donation than with 0 matching. 
With 2:1 matching, people have 0.48 percent more likelihood of responding to the donation than with 0 matching.
With 3:1 matching, people have 0.49 percent more likelihood of responding to the donation than with 0 matching.

2(c): 0.0048 - 0.0029 = 0.0019.

2(d): Having the donation matching offer will increase people's likelihood of making the donation. However, increasing the matching rate (from 1:1 to 2:1 to 3:1) does not induce a significant increase in people people's likelihood of making the donation, but we can still conclude that matching donations is useful in soliciting people to donate and greater matching ratio may still result in higher response rate (increase from intercept to ratio1 to ratio2 to ratio3). 

In [36]:
#3a
blue_states_df = df[df['red_state_dummy'] == 0]
red_states_df = df[df['red_state_dummy'] == 1]

X_blue = blue_states_df['treatment']
X_blue = sm.add_constant(X_blue)
y_blue = blue_states_df['donation_dummy']
model_blue = sm.OLS(y_blue, X_blue).fit()
print(model_blue.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.3567
Date:                Sat, 27 Jan 2024   Prob (F-statistic):              0.550
Time:                        15:22:09   Log-Likelihood:                 15783.
No. Observations:               29806   AIC:                        -3.156e+04
Df Residuals:                   29804   BIC:                        -3.155e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0200      0.001     14.085      0.0

In [37]:
X_red = red_states_df['treatment']
X_red = sm.add_constant(X_red)
y_red = red_states_df['donation_dummy']
model_red = sm.OLS(y_red, X_red).fit()
print(model_red.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     17.24
Date:                Sat, 27 Jan 2024   Prob (F-statistic):           3.30e-05
Time:                        15:22:51   Log-Likelihood:                 10838.
No. Observations:               20241   AIC:                        -2.167e+04
Df Residuals:                   20239   BIC:                        -2.166e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0146      0.002      8.398      0.0

Interpretation: For blue states, the intercept B0 is 0.02, which means that without any treatment, 2 percent of the control group will donate. B1 means that having the matching offer will increase people's likellihood of making the donation by 0.001 or 0.1 percent. For red states, the intercept B0 indicates that 1.46% of the control group will donate. Having the matching offer will increase people's likellihood of making the donation by 0.0088 or 0.88 percent. Hence, It seems to be more effective in the red state as the coefficient is larger, as the coefficnet for treatment associated with red states is statistically significant.  

3(b). The treatment coefficient does not have a causal interpretation in each of the two regressions. The difference in the treatment effect between states does not have a causal interpretation since it is not randomly assigned. 

In [40]:
#4a
X_4 = df['treatment']
X_4 = sm.add_constant(X_4)
y_4 = df['donation_amount']
model_4 = sm.OLS(y_4, X_4).fit()
print(model_4.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.463
Date:                Sat, 27 Jan 2024   Prob (F-statistic):             0.0628
Time:                        15:51:17   Log-Likelihood:            -1.7946e+05
No. Observations:               50082   AIC:                         3.589e+05
Df Residuals:                   50080   BIC:                         3.589e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.8133      0.067     12.063      0.0

Interpretation: B0 is 0.813, which means that for people without treatment(control), they are expected to donate 0.813 dollar. B1 = 0.154, indicating that making the matching offer will increase the amount of contribution by 0.154 dollar. The treatment coefficient does not have a causal interpretation as there could be situations where there's no correlation between the two variables since it contains zero in the interval with 95% confidence interval (-0.008, 0.315). 

In [41]:
#4b
donation_df = df[df['donation_dummy'] == 1]
X_5 = donation_df['treatment']
X_5 = sm.add_constant(X_5)
y_5 = donation_df['donation_amount']
model_5 = sm.OLS(y_5, X_5).fit()
print(model_5.summary())


                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    0.3374
Date:                Sat, 27 Jan 2024   Prob (F-statistic):              0.561
Time:                        15:56:01   Log-Likelihood:                -5326.8
No. Observations:                1034   AIC:                         1.066e+04
Df Residuals:                    1032   BIC:                         1.067e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         45.5403      2.423     18.792      0.0

B0 indicates that people are expected to donate 45.54 dollar for those who are willing to contribute. B1 indicates that for those who are willing to contribute some positive amount, having the matching offer will decrease the amount of contribution by $1.668. The treatment coefficient does not have a causal interpretation. 